### ISSUE TITLE
Audit new stackmap code for accessibility

### ISSUE THREAD

**Issue by kevinreiss** [pulibrary/orangelight#3564]
### Steps to recreate

1. Visit catalog-staging.princeton.edu
2. Make sure the firestone locator option in /features is OFF
3. Find any record with a stackmap button
4. Click the stackmap where to find it button
5. Modal rendering map will open. 

### URL
* Record Page: https://catalog-staging.princeton.edu/catalog/99118269913506421
* Search Results: https://catalog-staging.princeton.edu/catalog?utf8=%E2%9C%93&sort=score+desc%2C+pub_date_start_sort+desc%2C+title_sort+asc&f1=all_fields&f2=author&f3=title&op2=AND&op3=AND&q1=&q2=&q3=&f_inclusive%5Badvanced_location_s%5D%5B%5D=firestone%24nec&range%5Bpub_date_start_sort%5D%5Bbegin%5D=&range%5Bpub_date_start_sort%5D%5Bend%5D=&search_field=all_fields&q=cats

### Acceptance criteria
- [x] Check the revised button approach to "Where to find it"
- [x] Test the Modal that pops up. 
- [x] WCAG Conformance level AA 
- [x] I can successfully use this feature with Voice Over in Safari.
- [x] I have checked this feature using one of the following browser extensions and confirmed that it does not add any level A or level AA issues:
    * axe dev tools
    * Lighthouse
    * Wave

### Notes
The vendor has implemented the where to find it message as a button rather than a link.

**Comment by sandbergja** [comment]
@rladdusaw , @maxkadel , and I found 2 high priority accessibility issues, both related to color contrast.  We identified 3 other issues, which I think would be useful to raise with the vendor, but which I don't consider to be blockers.  [This spreadsheet details the issues we found](https://github.com/pulibrary/orangelight/files/11569524/Stackmap.accessibility.audit.xlsx).

**Comment by kevinreiss** [comment]
Will ask teh vendor to make these adjustments. Will ask for an outline of the button to deal with the high priority contrast issue.

**Comment by kevinreiss** [comment]
Request made 5/31.

**Comment by kevinreiss** [comment]
Can we re-test this? Vendor relayed to me that they updated the code per our requests.

**Comment by maxkadel** [comment]
As we've discussed, the new button colors (black and orange), although they pass A and AA contrast requirements, do not match the rest of the theme and are for some reason still difficult to read. 

As far as I can tell, they have not addressed the issue of the button border - from what I can see, it still does not have a border (all size values for the border are set to 0). I would appreciate being double checked on this, though.

**Comment by kevinreiss** [comment]
Confirm that the new stackmap button reads properly with a screenreader.

**Comment by kevinreiss** [comment]
Vendor tells me they've updated the button with a better color scheme. Can someone re-check this and close the ticket if it looks good.

**Comment by maxkadel** [comment]
Looks fine to me
![Button on search page](https://github.com/pulibrary/orangelight/assets/45948126/5954d932-36dc-4ed5-85d1-7b9cfe1e9b81)

![Button on record page](https://github.com/pulibrary/orangelight/assets/45948126/0a34f1e5-5ad7-4799-9f98-172a067d3ab0)
